packages/react-components/react-aria/library/src/useTypingAnnounce/useTypingAnnounce.ts

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import { useTimeout } from '@fluentui/react-utilities';
+import { useAnnounce, useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
+import type { AnnounceOptions } from '@fluentui/react-shared-contexts';
+import { AriaLiveAnnounceFn } from '../AriaLiveAnnouncer/AriaLiveAnnouncer.types';
+
+type Message = {
+  message: string;
+  options: AnnounceOptions;
+};
+
+const valueMutationOptions = {
+  attributes: true,
+  subtree: true,
+  characterData: true,
+  attributeFilter: ['value'],
+};
+
+export const useTypingAnnounce = (inputEl: React.RefObject<HTMLElement>): { typingAnnounce: AriaLiveAnnounceFn } => {
+  const { targetDocument } = useFluent();
+  const win = targetDocument?.defaultView;
+  const { announce } = useAnnounce();
+
+  const observer = React.useRef<MutationObserver>();
+  const [setTypingTimeout, clearTypingTimeout] = useTimeout();
+  const messageQueue = React.useRef<Message[]>([]);
+
+  const callback: MutationCallback = React.useCallback((mutationList, observer) => {
+    clearTypingTimeout();
+    setTypingTimeout(() => {
+      messageQueue.current.forEach(({ message, options }) => {
+        announce(message, options);
+      });
+      messageQueue.current = [];
```

Comment:
super nit. Since we're not relying on references to the array for anything, just clear it out rather than create a new array.

```suggestion
      messageQueue.current.length = 0;
```